### PR TITLE
Update "Changing to Remote H2" documentation

### PR DESCRIPTION
**Pull request by renuka-fernando**
fix wso2/docs-apim#624
- Update the documentation

### CODE REVIEW

File: en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md [hunk 1]
```diff
@@ -1,429 +1,147 @@
...
 
+## Setting up Remote H2
 
-### Setting up the database
-
-Download and install the H2 database engine on your computer.
-
-!!! info
-For instructions on installing DB2 Express-C, see [H2 installation guide.](http://www.h2database.com/html/quickstart.html)
+The following sections describe how to set up a MySQL database to replace the default H2 database in your WSO2 product:
```
Comment (npamudika) [inline en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md]: Let's change MySQL to remote H2 here

File: en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md [hunk 2]
```diff
@@ -1,429 +1,148 @@
...
-                <dataSource>jdbc/MY_DATASOURCE_NAME</dataSource>
-            </dbConfig>
+    ``` tab="Example"
+    [database.shared_db]
+    type = "h2"
+    url = "jdbc:h2:tcp://localhost/~/shared_db"
+    username = "regadmin"
+    password = "regadmin"
+    driver = "org.h2.Driver"
+    validationQuery = "SELECT 1"
```
Comment (vithu30) [inline en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md]: For H2, we do not need to specify driver and validation query. It is sufficient to specify type, url, username and password

File: en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md [hunk 3]
```diff
@@ -1,429 +1,148 @@
...
-H2 is not recommended in production
-
-The embedded H2 database is NOT recommended in enterprise testing and production environments. It has lower performance, clustering limitations, and can cause file corruption failures. Please use an industry-standard RDBMS such as Oracle, PostgreSQL, MySQL, or MS SQL instead.
-
-You can use the embedded H2 database in development environments and as the local registry in a registry mount.
+### Creating the datasource connection to Remote H2
 
-!!! tip
-Before you begin
+A datasource is used to establish the connection to a database. By default, `WSO2_SHARED_DB` and `WSO2AM_DB` datasource are configured in the `deployment.toml` file for the purpose of connecting to the default H2 databases.
```
Comment (npamudika) [inline en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md]: Shall we change like below.
```suggestion
A datasource is used to establish the connection to a database. By default, `WSO2_SHARED_DB` and `WSO2AM_DB` datasource are configured in the `deployment.toml` file for the purpose of connecting to the default embedded H2 databases.
```

File: en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md [hunk 4]
```diff
@@ -1,429 +1,148 @@
...
 
--   Set up the database as explained in [Setting up Remote H2](https://docs.wso2.com/display/ADMIN44x/Setting+up+Remote+H2) .
--   Download the H2 database driver and copy it to WSO2 IoT Server by following the steps under [Setting up drivers](https://docs.wso2.com/display/ADMIN44x/Setting+up+Remote+H2#SettingupRemoteH2-Settingupdrivers) .
+After setting up the Remote H2 database to replace the default H2 database, either change the default configurations of the `WSO2_SHARED_DB` and `WSO2AM_DB` datasource, or configure a new datasource to point it to the new database as explained below.
```
Comment (npamudika) [inline en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md]: Shall we change "Remote" to "remote"?

File: en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md [hunk 5]
```diff
@@ -1,429 +1,148 @@
...
+    **If you are configuring API-M in a distributed setup** , do the changes in all the WSO2 API-M components.
 
-### Setting up datasource configurations
+Follow the steps below to change the type of the default datasource.
 
-A datasource is used to establish the connection to a database. By default, `WSO2_CARBON_DB` datasource is used to connect to the default  H2 database, which stores registry and user management data. After setting up the Remote H2 database to replace the default H2 database, either [change the default configurations of the `WSO2_CARBON_DB` datasource](#ChangingtoRemoteH2-Changingthedefaultdatabase) , or [configure a new datasource](#ChangingtoRemoteH2-Configuringnewdatasourcestomanageregistryorusermanagementdata) to point it to the new database as explained below.
+1.  Open the `<API-M_HOME>/repository/conf/deployment.toml` configuration file and locate the `[database.shared_db]` and `[database.apim_db]` configuration elements.
 
-#### Changing the default WSO2\_CARBON\_DB datasource
+1.  You simply have to update the URL pointing to your MySQL database, the username, and password required to access the database and the MySQL driver details as shown below.
```
Comment (npamudika) [inline en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md]: Need to change Mysql to H2

File: en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md [hunk 6]
```diff
@@ -1,429 +1,148 @@
...
+    type = "h2"
+    url = "jdbc:h2:tcp://localhost/~/shared_db"
+    username = "regadmin"
+    password = "regadmin"
+    driver = "org.h2.Driver"
+    validationQuery = "SELECT 1"
+
+    [database.apim_db]
+    type = "h2"
+    url = "jdbc:h2:tcp://localhost/~/apim_db"
```
Comment (npamudika) [inline en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md]: Don't we need to specify the port, if we connects to a remote h2 database server?

Comment (renuka-fernando) [reply en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md]: Updated the DB url via https://github.com/wso2/docs-apim/pull/627/commits/a6c4241898fd5d160a5733eb6966e177f995dc51

File: en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md [hunk 7]
```diff
@@ -1,429 +1,135 @@
...
-
-    -   For Linux: `<PRODUCT_HOME>/bin/wso2server.sh -Dsetup`
-
-        !!! warning
-        Deprecation of -DSetup
-    When proper Database Administrative (DBA) practices are followed, the systems (except analytics products) are not granted DDL (Data Definition) rights on the schema. Therefore, maintaining the `-DSetup` option is redundant and typically unusable. **As a result, from [January 2018 onwards](https://wso2.com/products/carbon/release-matrix/) WSO2 has deprecated the `-DSetup` option** . Note that the proper practice is for the DBA to run the DDL statements manually so that the DBA can examine and optimize any DDL statement (if necessary) based on the DBA best practices that are in place within the organization.
 
+1.  To create tables in the registry and user manager database (`WSO2_SHARED_DB`) use the script: `<API-M_HOME>/dbscripts/h2.sql`
 
+1.  To create tables in the registry and user manager database (`WSO2_SHARED_DB`) use the script: `<API-M_HOME>/dbscripts/apimgt/h2.sql`
```
Comment (malinthaprasan) [inline en/docs/InstallAndSetup/SettingUpDatabases/ChangingDefaultDatabases/changing-to-remote-h2.md:33]: This has few typos but seems this doc is taken out from our documentation? if so we don't need to fix. @renuka-fernando

**Pull request merged**